students/M_Sunday/lesson08/circle.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+from math import pi
+
+
+class Circle(object):
+
+    @staticmethod
+    def entry_check(the__radius):
+        if isinstance(the__radius, (str, list, tuple, dict)):
+            raise TypeError("Radius/Diameter entry must be a single, positive, and "
+                            "non-string value")
+        else:
+            if the__radius < 0:
+                raise ValueError("Radius/Diameter must be greater than 0")
+            else:
+                return the__radius
+
+    def __init__(self, the_radius):
+        self._radius = self.entry_check(the_radius)
+
+    def __repr__(self):
+        return "Circle({})".format(self._radius)
+
+    def __str__(self):
+        return "Circle with radius: {0:.6f}".format(float(self._radius))
+
+    def __add__(self, other):
+        new_circle = self._radius + other._radius
```

Comment:
In Python a leading underscore conventionally indicates a 'protected' attribute that should only be accessed within a class or subclass